text/todo-generic-payload.md

```diff
@@ -0,0 +1,583 @@
+| ----- | ---------------------------------------- | -------------------------------------------------------------------------- |
+| 0     | GENERIC_PAYLOAD_FUNCTION_TYPE_LOGICAL    | Logical state control (true/false, on/off, high/low).                      |
+| 1     | GENERIC_PAYLOAD_FUNCTION_TYPE_CONTINUOUS | Continuous value control (e.g., servo angle, motor speed). Uses min/max.   |
+| 2     | GENERIC_PAYLOAD_FUNCTION_TYPE_DISCRETE   | Discrete value selection (e.g., mode number, PWM frequency). Uses min/max. |
+| 3     | GENERIC_PAYLOAD_FUNCTION_TYPE_BITMASK    | Bitmask value selection                                                    |
+
+#### GENERIC_PAYLOAD_ERROR_FLAGS
+
+Error flags for a generic payload. This is a bitmask enumeration.
+
+| Value | Name                                       | Description                                                                                                              |
+| ----- | ------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------ |
+| 1     | GENERIC_PAYLOAD_ERROR_FLAGS_SOFTWARE_ERROR | There is an error with the generic payload's software.                                                                   |
+| 2     | GENERIC_PAYLOAD_ERROR_FLAGS_HARDWARE_ERROR | There is an error with the generic payload's hardware.                                                                   |
+| 4     | GENERIC_PAYLOAD_ERROR_FLAGS_OVERTEMP_FAULT | A specified limit has been reached by the generic payload's temperature sensor. Applies if payload monitors temperature. |
+| 8     | GENERIC_PAYLOAD_ERROR_FLAGS_CUSTOM         | Generic payload custom failure. See custom error flag bitmask for details.                                               |
+
+### Messages
+
+#### GENERIC_PAYLOAD_DESCRIPTION (Message ID: 59999)
+
+Message broadcasted at a low rate (nominally < 1 Hz or on request) to announce the payload's presence and capabilities.
+
+| Field Name             | Type        | Description                                                          |
+| ---------------------- | ----------- | -------------------------------------------------------------------- |
+| payload_id             | uint8_t     | Component ID of the generic payload.                                 |
+| num_functions          | uint16_t    | Number of functions this payload provides.                           |
+| num_telemetry_channels | uint16_t    | Number of telemetry channels this payload provides.                  |
+| name                   | char[32]    | Name of the generic payload for UI display (NULL-terminated string). |
+| mass                   | uint16_t    | Mass of the payload in grams. 0 if not available.                    |
+| torque_arm             | uint16_t[3] | Torque arm in Payload Frame (mm)                                     |
```

Comment:
I don't know what this means. Can it ever be not-available or invalid?

Reply:
yeah this is nuanced one that may not be relevant, i think the idea was to describe the CoM relative to the vehicles CoM or its Coordinate frame